src/libs/decoders/opus.c

```diff
@@ -0,0 +1,662 @@
+#endif
+
+#ifdef _MSC_VER
+// Avoid warning about getenv()
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include <stdlib.h> // getenv()
+
+// On macOS with GCC, pkg-config only include the opus/ subdirectory
+// itself instead of the parent, so we take this into account:
+#if defined(MACOSX) && ! defined(__clang__) && defined(__GNUC__)
+#include <opusfile.h>
+#else
+#include <opus/opusfile.h>
+#endif
+#include <speex/speex_resampler.h>
+
+#include "SDL_sound.h"
+#define __SDL_SOUND_INTERNAL__
+#include "SDL_sound_internal.h"
+
+// The minimum buffer samples per channel: 120 ms @ 48 samples/ms, defined by opus
+#define OPUS_MIN_BUFFER_SAMPLES_PER_CHANNEL 5760
+
+// Opus's internal sample rates, to which all encoded streams get resampled
+#define OPUS_SAMPLE_RATE 48000
+#define OPUS_SAMPLE_RATE_PER_MS 48
+
+static Sint32 opus_init   (void);
+static void   opus_quit   (void);
+static Sint32 opus_open   (Sound_Sample* sample, const char* ext);
+static void   opus_close  (Sound_Sample* sample);
+static Uint32 opus_read   (Sound_Sample* sample);
+static Sint32 opus_rewind (Sound_Sample* sample);
+static Sint32 opus_seek   (Sound_Sample* sample, const Uint32 ms);
+
+static const char* extensions_opus[] = { "OPUS", NULL };
+
+const Sound_DecoderFunctions __Sound_DecoderFunctions_OPUS =
+{
+	{
+		extensions_opus,
+		"Ogg Opus audio using libopusfile",
+		"Kevin R Croft <krcroft@gmail.com>",
+		"https://www.opus-codec.org/"
+	},
+
+	opus_init,   /*   init() method */
+	opus_quit,   /*   quit() method */
+	opus_open,   /*   open() method */
+	opus_close,  /*  close() method */
+	opus_read,   /*   read() method */
+	opus_rewind, /* rewind() method */
+	opus_seek    /*   seek() method */
+};
+
+
+// Our private-decoder structure where we hold the opusfile, resampler,
+// circular buffer, and buffer tracking variables.
+typedef struct
+{
+	Uint64 of_pcm;                  // absolute position in consumed Opus samples
+	OggOpusFile* of;                // the actual opusfile we open/read/seek within
+	opus_int16* buffer;             // pointer to the start of our circular buffer
+	SpeexResamplerState* resampler; // pointer to an instantiated resampler
+	float rate_ratio;               // OPUS_RATE (48KHz) divided by desired sample rate
+	Uint16 buffer_size;             // maximum number of samples we can hold in our buffer
+	Uint16 decoded;                 // number of samples decoded in our buffer
+	Uint16 consumed;                // number of samples consumed in our buffer
+	Uint16 frame_size;              // number of samples decoded in one opus frame
+	SDL_bool eof;                   // indicates if we've hit end-of-file decoding
+} opus_t;
+
+
+static Sint32 opus_init(void)
+{
+	SNDDBG(("Opus init:              done\n"));
+	return 1; /* always succeeds. */
+} /* opus_init */
+
+
+static void opus_quit(void){
+	SNDDBG(("Opus quit:              done\n"));
+} // no-op
+
+
+/*
+ * Read-Write Ops Read Callback Wrapper
+ * ====================================
```

Comment:
Get rid of `=====…` in comment formatting, please.

Reply:
Snipped.  But there are still plenty in the builtin code.. is there a way you can work around those?  Given these are all git-based projects, presumably it's not unreadable to ask them to clean them up too? (and this has nothing to do with our personal preferences.. the use of === directly conflicts with git's BEFORE/AFTER conflict identifier; so I think this should be new universal cleanup in all code managed in git).

dr_flac.h:9
dr_mp3.h:12
dr_wav.h:10
stb.h:2
xxh3.h:15
xxhash.c:8
xxhash.h:8

Reply:
Fixed in https://github.com/dreamer/dosbox-staging/commit/5f71f3a6f1225306ff7fb7bccd96fd5e315db5df

Reply:
Flagged upstream too.. for what it's worth!
 dr_libs: https://github.com/mackron/dr_libs/issues/70
 stb: https://github.com/nothings/stb/issues/830
 xxHash: https://github.com/Cyan4973/xxHash/issues/277